### PR TITLE
🫓 Flatten analytics object on config

### PR DESCRIPTION
**Pull request by rowanc1**

### CODE REVIEW

**Comment by rowanc1** [comment]
@fwkoch looks like this will need the release of `myst-config`.

File: packages/site/src/pages/Article.tsx [hunk 1]
```diff
@@ -5,7 +5,7 @@ import { ErrorProjectNotFound } from './ErrorProjectNotFound';
 import type { PageLoader } from '../types';
 
 export function ArticlePage({ article }: { article: PageLoader }) {
-  const { hide_title_block, hide_footer_links } = article.frontmatter?.design ?? {};
+  const { hide_title_block, hide_footer_links } = (article.frontmatter as any)?.design ?? {};
```
Comment (fwkoch) [inline packages/site/src/pages/Article.tsx:8]: Some of this page-level customization may be a little kluge-y until we get site template option overrides in page frontmatter...

Comment (rowanc1) [reply packages/site/src/pages/Article.tsx:8]: Yep, can make it better as we go.

**Comment by rowanc1** [comment]
Do you recall if we put `References` in myst-config?

This is the only line left:

```
import type { References } from '@curvenote/site-common';
```

**Comment by fwkoch** [comment]
That type is the only thing left in site common...
```
import type { Root } from 'mdast';
import type { FootnoteDefinition } from 'myst-spec';

export type Citations = {
  order: string[];
  data: Record<string, { html: string; number: number; doi: string | undefined }>;
};

export type Footnotes = Record<string, FootnoteDefinition>;

export type References = {
  cite?: Citations;
  footnotes?: Footnotes;
  article?: Root;
};
```

Those classes are also basically duplicated in `myst-cli` and `myst-transforms`.

So if we move that anywhere, we can kill `site-common` entirely. Maybe we put it in `myst-common`? That package already has the two deps imported here...?

**Pull request merged**